patches/systemd-utils/0001-Add-managarm-support.patch

```diff
@@ -0,0 +1,906 @@
+ 
+         xsprintf(path, "/proc/self/fdinfo/%i", fd);
+ 
++        return -ENOTTY;
++
+         r = read_full_virtual_file(path, &fdinfo, NULL);
+         if (r == -ENOENT) /* if fdinfo doesn't exist we assume the process does not exist */
+                 return proc_mounted() > 0 ? -EBADF : -ENOSYS;
+diff --git a/src/basic/raw-clone.h b/src/basic/raw-clone.h
+index 6de67ab..7fd7164 100644
+--- a/src/basic/raw-clone.h
++++ b/src/basic/raw-clone.h
+@@ -7,7 +7,7 @@
+ 
+ #include <errno.h>
+ #include <sched.h>
+-#include <sys/syscall.h>
++// #include <sys/syscall.h>
+ 
+ #include "log.h"
+ #include "macro.h"
+@@ -74,6 +74,8 @@ static inline pid_t raw_clone(unsigned long flags) {
+                 } else
+                         ret = in_child ? 0 : child_pid;
+         }
++#elif defined(__managarm__)
++        assert(!"unimplemented raw_clone");
+ #else
+         ret = (pid_t) syscall(__NR_clone, flags, NULL);
+ #endif
+diff --git a/src/basic/raw-reboot.h b/src/basic/raw-reboot.h
+index e6bff30..08d9ee9 100644
+--- a/src/basic/raw-reboot.h
++++ b/src/basic/raw-reboot.h
+@@ -3,12 +3,16 @@
+ 
+ #include <linux/reboot.h>
+ #include <sys/reboot.h>
+-#include <sys/syscall.h>
++// #include <sys/syscall.h>
+ 
+ /* glibc defines the reboot() API call, which is a wrapper around the system call of the same name, but without the
+  * extra "arg" parameter. Since we need that parameter for some calls, let's add a "raw" wrapper that is defined the
+  * same way, except it takes the additional argument. */
+ 
+ static inline int raw_reboot(int cmd, const void *arg) {
++#if defined(__managarm__)
++        assert(!"unimplemented");
++#else
+         return (int) syscall(SYS_reboot, LINUX_REBOOT_MAGIC1, LINUX_REBOOT_MAGIC2, cmd, arg);
++#endif
+ }
+diff --git a/src/basic/socket-util.h b/src/basic/socket-util.h
+index c784125..1b23bc0 100644
+--- a/src/basic/socket-util.h
++++ b/src/basic/socket-util.h
+@@ -2,11 +2,11 @@
+ #pragma once
+ 
+ #include <inttypes.h>
++#include <netinet/in.h>
+ #include <linux/netlink.h>
+ #include <linux/if_ether.h>
+ #include <linux/if_infiniband.h>
+ #include <linux/if_packet.h>
+-#include <netinet/in.h>
+ #include <stdbool.h>
+ #include <stddef.h>
+ #include <string.h>
+diff --git a/src/core/exec-invoke.c b/src/core/exec-invoke.c
+index 56df5cf..f72eccf 100644
+--- a/src/core/exec-invoke.c
++++ b/src/core/exec-invoke.c
+@@ -3,6 +3,7 @@
+ #include <linux/sched.h>
+ #include <sys/eventfd.h>
+ #include <sys/ioctl.h>
++#include <sys/file.h>
+ #include <sys/mount.h>
+ #include <sys/prctl.h>
+ 
+diff --git a/src/libsystemd/sd-event/sd-event.c b/src/libsystemd/sd-event/sd-event.c
+index 64825ca..6685aa7 100644
+--- a/src/libsystemd/sd-event/sd-event.c
++++ b/src/libsystemd/sd-event/sd-event.c
+@@ -1891,7 +1891,7 @@ _public_ int sd_event_trim_memory(void) {
+ 
+         usec_t before_timestamp = now(CLOCK_MONOTONIC);
+         hashmap_trim_pools();
+-        r = malloc_trim(0);
++        r = 0;
```

Comment:
Why is this done?

Reply:
Looking at the man page, we can just stub this? Seems to be some sort of memory usage reduction item so stubbing inside mlibc should be fine.